cmd/destroy/build_control.go

```diff
@@ -42,27 +42,38 @@ func newBuildCtrl(name string, analyticsTracker, insights buildTrackerInterface,
 }
 
 type builderInterface interface {
-	GetServicesToBuildForImage(ctx context.Context, manifest *model.Manifest, findImg model.ImageFromManifest) ([]string, error)
+	GetSvcToBuildFromRegex(manifest *model.Manifest, imgFinder model.ImageFromManifest) (string, error)
+	GetServicesToBuildDuringDeploy(ctx context.Context, manifest *model.Manifest, svcsToDeploy []string) ([]string, error)
 	Build(ctx context.Context, options *types.BuildOptions) error
 }
 
 func (bc buildCtrl) buildImageIfNecessary(ctx context.Context, manifest *model.Manifest) error {
 	oktetoLog.Debug("checking if destroy.image is already built")
-	svcsToBuild, err := bc.builder.GetServicesToBuildForImage(ctx, manifest, func(manifest *model.Manifest) string {
+	svcToBuild, err := bc.builder.GetSvcToBuildFromRegex(manifest, func(manifest *model.Manifest) string {
 		return manifest.Destroy.Image
 	})
 	if err != nil {
-		return err
+		if errors.Is(err, buildv2.ErrOktetBuildSyntaxImageIsNotInBuildSection) {
+			return fmt.Errorf("the image '%s' is not in the build section of the manifest", svcToBuild)
+		}
+		oktetoLog.Debugf("error getting services to build for image '%s': %s", svcToBuild, err)
 	}
-	if len(svcsToBuild) < 1 {
+
+	svcsToBuild, err := bc.builder.GetServicesToBuildDuringDeploy(ctx, manifest, []string{svcToBuild})
+	if err != nil {
+		return fmt.Errorf("failed to get services to build: %w", err)
+	}
+
+	if len(svcsToBuild) == 0 {
+		oktetoLog.Debug("destroy.image is already built")
 		return nil
 	}
 	buildOptions := &types.BuildOptions{
 		EnableStages: true,
 		Manifest:     manifest,
 		CommandArgs:  svcsToBuild,
```

Comment:
Should then specify here `svcToBuild`? Or should we just update the info logger below to use `svcsToBuild` instead of `svcToBuild`?